extended/src/test/java/io/kubernetes/client/extended/kubectl/KubectlApplyTest.java

```diff
@@ -98,4 +106,62 @@ public void testApplyConfigMap() throws KubectlException, IOException {
         1, patchRequestedFor(urlPathEqualTo("/api/v1/namespaces/foo/configmaps/bar")));
     assertNotNull(configMap);
   }
+
+  @Test
+  public void testApplyServiceInRancher() throws KubectlException, IOException {
```

Comment:
@dddddai this repo will stay vendor-neutral so i dont think it's a good idea to put such a test that verifying the behavior for rancher. 

a better way to do this is to have an additional setter e.g. `setAPIPrefix` for the class `GenericKubernetesApi` so that users can tweak api path freely.

Reply:
@yue9944882 Thanks for your review

> this repo will stay vendor-neutral so i dont think it's a good idea to put such a test that verifying the behavior for rancher.

Actually this test just makes sure `tweakCallForCoreV1Group` work fine with api prefix, so I guess the function name should be changed to something like `testApplyServiceWithApiPrefix`?

> a better way to do this is to have an additional setter e.g. setAPIPrefix for the class GenericKubernetesApi so that users can tweak api path freely.

But users could not see `GenericKubernetesApi` when using `Kubectl`, for example:
```java
V1Service service =
    Kubectl.apply(V1Service.class)
        .apiClient(new ClientBuilder().setBasePath("http://localhost:" + 8384 + prefix).build())
        .resource(
            new V1Service()
                .kind("Service")
                .apiVersion("v1")
                .metadata(
                    new V1ObjectMeta()
                        .namespace("foo")
                        .name("bar")
                        .labels(Collections.singletonMap("app", "bar")))
                .spec(spec))
        .execute();
```
In this case, users can only set api prefix on `ApiClient`, shall I add the setter in `ApiClient`?

Reply:
@dddddai IIUC this pull doesnt need to involve changes in the `KubectlApply` related sources. making `GenericKuberentesApi` able to respect the prefix from the `ApiClient` should be sufficient. so we can remove this test, are we?